### PR TITLE
Fix warning on conversion of anon func with implicit parameters

### PR DESCRIPTION
**Pull request by jcouv**
Fixes https://github.com/dotnet/roslyn/issues/59336

### CODE REVIEW

File: src/Compilers/CSharp/Portable/BoundTree/UnboundLambda.cs [hunk 1]
```diff
@@ -517,6 +518,7 @@ internal UnboundLambdaState WithCaching(bool includeCache)
         public UnboundLambda UnboundLambda => _unboundLambda;
 
         public abstract MessageID MessageID { get; }
+        public abstract bool HasParameterNames();
```
Comment (cston) [inline src/Compilers/CSharp/Portable/BoundTree/UnboundLambda.cs:521]: >HasParameterNames() [](http://example.com/codeflow?start=29&length=19)

Is this distinct from `HasNames` or `HasSignature`?

File: src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs [hunk 2]
```diff
@@ -155949,5 +155949,26 @@ void M()
                 Diagnostic(ErrorCode.WRN_NullabilityMismatchInReturnTypeOfTargetDelegate, "d2").WithArguments("ref readonly string? D<string?>.Invoke()", "D<string>").WithLocation(9, 28)
                 );
         }
+
+        [Fact, WorkItem(59336, "https://github.com/dotnet/roslyn/issues/59336")]
+        public void ParameterNameCrash()
```
Comment (cston) [inline src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs:155954]: Consider changing the test name to be more specific. Perhaps `AnonymousMethodNoSignatureAndDelegateAttributes`?

File: src/Compilers/CSharp/Portable/BoundTree/UnboundLambda.cs (hunk 1 above)
Comment (jcouv) [reply src/Compilers/CSharp/Portable/BoundTree/UnboundLambda.cs]: Looks like we have a triplicate. I'll remove two

**Review by cston — APPROVED** [review]

File: src/Compilers/CSharp/Portable/BoundTree/UnboundLambda.cs [hunk 3]
```diff
@@ -1357,7 +1356,6 @@ internal sealed class PlainUnboundLambdaState : UnboundLambdaState
             _isStatic = isStatic;
         }
 
-        public override bool HasNames { get { return !_parameterNames.IsDefault; } }
```
Comment (RikkiGibson) [inline src/Compilers/CSharp/Portable/BoundTree/UnboundLambda.cs:1360]: so this member was just totally redundant in practice?

**Review by RikkiGibson — APPROVED** [review]

File: src/Compilers/CSharp/Portable/BoundTree/UnboundLambda.cs (hunk 3 above)
Comment (jcouv) [reply src/Compilers/CSharp/Portable/BoundTree/UnboundLambda.cs:1360]: Yup, they have the same meaning. We had `HasNames`, `HasSignature` and I was adding yet another one...

**Pull request merged**